pkg/asset/imagebased/image/imagebased_config.go

```diff
@@ -0,0 +1,409 @@
+		allErrs = append(allErrs, err...)
+	}
+	if err := i.validateProxy(); err != nil {
+		allErrs = append(allErrs, err...)
+	}
+
+	return allErrs
+}
+
+func (i *ImageBasedInstallationConfig) validatePullSecret() field.ErrorList {
+	var allErrs field.ErrorList
+
+	pullSecretPath := field.NewPath("pullSecret")
+
+	if i.Config.PullSecret == "" {
+		allErrs = append(allErrs, field.Required(pullSecretPath, "you must specify a pullSecret"))
+		return allErrs
+	}
+
+	if err := validate.ImagePullSecret(i.Config.PullSecret); err != nil {
+		allErrs = append(allErrs, field.Invalid(pullSecretPath, i.Config.PullSecret, err.Error()))
+	}
+
+	return allErrs
+}
+
+func (i *ImageBasedInstallationConfig) validateSSHKey() field.ErrorList {
+	var allErrs field.ErrorList
+
+	sshKeyPath := field.NewPath("sshKey")
+
+	// empty SSHKey is fine
+	if i.Config.SSHKey == "" {
+		return nil
+	}
+
+	if err := validate.SSHPublicKey(i.Config.SSHKey); err != nil {
+		allErrs = append(allErrs, field.Invalid(sshKeyPath, i.Config.SSHKey, err.Error()))
+	}
+
+	return allErrs
+}
+
+func (i *ImageBasedInstallationConfig) validateAdditionalTrustBundle() field.ErrorList {
+	var allErrs field.ErrorList
+
+	additionalTrustBundlePath := field.NewPath("additionalTrustBundle")
+
+	// empty AdditionalTrustBundle is fine
+	if i.Config.AdditionalTrustBundle == "" {
+		return nil
+	}
+
+	if err := validate.CABundle(i.Config.AdditionalTrustBundle); err != nil {
+		allErrs = append(allErrs, field.Invalid(additionalTrustBundlePath, i.Config.AdditionalTrustBundle, err.Error()))
+	}
+
+	return allErrs
+}
+
+func (i *ImageBasedInstallationConfig) validateSeedImage() field.ErrorList {
+	var allErrs field.ErrorList
+
+	seedImagePath := field.NewPath("seedImage")
+
+	if i.Config.SeedImage == "" {
+		allErrs = append(allErrs, field.Required(seedImagePath, "you must specify a seedImage"))
+	}
+
+	return allErrs
+}
+
+func (i *ImageBasedInstallationConfig) validateSeedVersion() field.ErrorList {
+	var allErrs field.ErrorList
+
+	seedVersionPath := field.NewPath("seedVersion")
+
+	if i.Config.SeedVersion == "" {
+		allErrs = append(allErrs, field.Required(seedVersionPath, "you must specify a seedVersion"))
+	}
+
+	return allErrs
+}
+
+func (i *ImageBasedInstallationConfig) validateInstallationDisk() field.ErrorList {
+	var allErrs field.ErrorList
+
+	installationDiskPath := field.NewPath("installationDisk")
+
+	if i.Config.InstallationDisk == "" {
+		allErrs = append(allErrs, field.Required(installationDiskPath, "you must specify an installationDisk"))
+	}
+
+	return allErrs
+}
+
+func (i *ImageBasedInstallationConfig) validateNetworkConfig() field.ErrorList {
+	var allErrs field.ErrorList
+
+	networkConfig := field.NewPath("networkConfig")
+
+	staticNetworkConfigGenerator := staticnetworkconfig.New(logrus.StandardLogger(), staticnetworkconfig.Config{MaxConcurrentGenerations: 2})
+
+	// empty NetworkConfig is fine
+	if i.Config.NetworkConfig.String() == "" {
```

Comment:
nit: maybe function can start with this, without need to create staticNetworkConfigGenerator in case there is no nm config?
But this is really super nit, just to consider

Reply:
I agree, I have update this check, it's a lot cleaner this way :+1: